### ISSUE TITLE
Migration to version 9.0

### ISSUE THREAD

**Issue by pedrobaeza** [OCA/vertical-medical#50]
# Todo

https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0
# Modules to migrate
- [x] medical - @lasley https://github.com/OCA/vertical-medical/pull/69
- [ ] medical_family
- [ ] medical_insurance - #102
- [ ] medical_insurance_us - @lasley
- [x] medical_medicament - #74
- [ ] medical_medicament_attributes - @lasley
- [x] medical_medication - #75 
- [ ] medical_medication_us - #76
- [x] medical_disease - #71 (Renamed to medical_pathology) 
- [x] medical_patient_disease - #72 
- [ ] medical_patient_ethnicity - @lasley
- [ ] medical_patient_occupation - @lasley
- [x] medical_prescription - #79 
- [x] medical_pharmacy - #77 
- [ ] medical_pharmacy_us - #78 
- [ ] medical_prescription_thread - @lasley 
- [ ] medical_his
- [ ] medical_emr
- [ ] medical_emr_data
- [ ] medical_operational
- [ ] omedical_genetics
- [ ] oemedical_gynecology_and_obstetrics
- [ ] oemedical_icu
- [ ] oemedical_invoice
- [x] oemedical_lab - (renamed to medical_lab) - #137 
- [ ] oemedical_lifestyle
- [ ] oemedical_pediatrics
- [ ] oemedical_socioeconomics
- [ ] oemedical_surgery
- [ ] web_doc_oemedical

**Comment by lasley** [comment]
Hi All - How are we going to handle this? I vote this gets put on the afterburner until we move 8.0 incubator into 8.0.

**Comment by JonathanNEMRY** [comment]
I agree with @lasley : first we have to move 8.0-incubator into 8.0 before starting 9.0

**Comment by lasley** [comment]
Sweet, glad we're on the same page @JonathanNEMRY. Looks like we're getting close to the finish line on the 8.0-incubator too :smile:

**Comment by Kickball** [comment]
What is the progress on migrating to 8.0?

**Comment by lasley** [comment]
As far as I know, 0%. Working on a pharmacy module on my end at the moment, but that's in 8.0.

I'm hoping to start tackling 9.0 soon, but we would definitely need to make sure that the >10 PRs currently pending are finalized before even attempting that

**Comment by Kickball** [comment]
How close are those PRs to being ready?

I'm considering deploying Vertical-Medical, but I feel it might be best to wait if 8.0 will be finished within less than a month.

**Comment by lasley** [comment]
They're all ready IIRC, just need to be reviewed and merged. My team is internally targeting mid-Feb for 9.0

**Comment by Kickball** [comment]
I'll wait until February then, thanks!

**Comment by lasley** [comment]
FYI now that 8.0 PR is submitted, we've planned out what we need for 9.0. 

My team will take on the following modules, and should be done in the next few weeks (some are already done):
- medical (this is being split into multiple modules)
- medical_family
- medical_insurance
- medical_insurance_us
- medical_medicament
- medical_medicament_attributes
- medical_medication
- medical_medication_us
- medical_disease
- medical_patient_disease
- medical_patient_ethnicity
- medical_patient_occupation
- medical_prescription
- medical_pharmacy
- medical_pharmacy_us
- medical_prescription
- medical_prescription_thread

The following modules are not within the scope of our current project, so if someone else would like to coordinate on them let me know (we're open for volunteers on the above too :stuck_out_tongue_closed_eyes: ) - My team likely won't be touching them for a while:
- medical_his
- medical_emr - a lot of this has already been stripped out into other modules, but there's still some left
- medical_emr_data - needs to be broken down into the independent modules
- medical_operational
- omedical_genetics
- oemedical_gynecology_and_obstetrics
- oemedical_icu
- oemedical_invoice 
- oemedical_lab
- oemedical_lifestyle
- oemedical_pediatrics
- oemedical_socioeconomics
- oemedical_surgery
- web_doc_oemedical

@pedrobaeza -

Would you mind updating your 9.0 conversion requirements for the new module layout above?

Additionally - In the interest of being lazy and not re-resolving all the conflicts, I propose that we recreate the 9.0 branch from 8.0 once merged, then just re-run the upgrade script. As far as I can tell, that's the only change there. Maybe other suggestions?

**Comment by pedrobaeza** [comment]
OK, I have already updated the issue, and please ping when your PRs are merged on 8.0 branch to forward copy to 9.0 before starting migration.

**Comment by lasley** [comment]
Awesome thank you!

**Comment by JayVora-SerpentCS** [comment]
We are adding team to speed this up!

**Comment by lasley** [comment]
@JayVora-SerpentCS - Awesome, let's coordinate on this then. 

PRs are currently pending for about half of the ones I listed in my comment above as being completed on my end, which I believe cover all dependencies for the rest of the modules.

You looking to start anywhere in specific, or just looking to get everything to 9.0 as a whole?

As a general note, the modules with `oemedical` prefix have not yet been touched as part of this massive dev push (to 8.0). Anything with `medical` prefix should be fairly up to date, but is likely lacking in test coverage.

**Comment by JayVora-SerpentCS** [comment]
Guess, we should make v8 modules completed first.

**Comment by lasley** [comment]
@JayVora-SerpentCS Honestly I was just going to skip v8 on the ones that we didn't port, but that would definitely be a good start. 

The modules with `oemedical` prefix need a significant amount of work before they are anything close to ready for production. The ones with `medical` prefix should be somewhat good, but do not fall in line with OCA standards in a lot of ways (particularly modularity & naming).

Basically everything is 0% test coverage as well. We have made an internal goal towards 90% coverage on v9.

**Comment by lasley** [comment]
I would say that if you're going to just flat out upgrade a module without restructuring, please only bring it to v8. We can then use v9 as a staging for truly good modules. This library has a significant amount of errors as is, and will require a lot of QA to actually upgrade.

**Comment by askz** [comment]
Hi folks, and thanks for your hard work on this project!
I extended the appointment module to generate invoice automaticaly when app is done, for one of my clients. Would you be interested on this? I don't want to waste your time.

**Comment by pedrobaeza** [comment]
Yeah, it can be an interesting module. You can make a PR for it.

**Comment by lasley** [comment]
@askz There is already a medical_invoice module, does this by any chance upgrade any of that logic?

**Comment by lasley** [comment]
(also any contribution is welcome IMO)

**Comment by pedrobaeza** [comment]
Of course, that's the spirit of OCA! It can follow 2 approaches:
- Open an issue to discuss the convenience of the new feature/module plus a way to implement it (if it's no done)
- Open a PR and discuss over the work already done how to improve it/complement it with the rest of the OCA ecosystem.

It's convenient though to take into account if there's already an existing feature about it, because it's not allowed in OCA by default 2 modules that covers the same feature.

**Comment by lmignon** [comment]
@lasley an invoice is an invoice IMO, invoice must stay an account_invoice and we should avoid to define a new model for this transverse concept.

**Comment by askz** [comment]
@lasley yup. Actually I've taken the snippet that generate the invoice and invoking it when stage is 'done' : 
![screenshot](https://cloud.githubusercontent.com/assets/854038/13181853/4f0b9b26-d72f-11e5-86c9-b8a2981784fc.png)

**Comment by askz** [comment]
@lmignon you're totally right, and I haven't done that. Just generating account.invoice when an appointment is done ;) Sorry for the spam, I'll open an issue when my code will be reviewed a little.

**Comment by lasley** [comment]
@lmignon That makes sense, I think I mis-posed my question. Really I was asking if the logic had been moved from the invoice module, or if it was new. 

Part of the direction of my inquiry was actually due to the logic I had seen in the pre-existing library, and the fact that it did not use the account.invoice core. 

@askz Any way to switch this a bit so that it's using account.invoice instead?

**Comment by askz** [comment]
@lasley I edited my last comment. And this is what I did. But I'm constrained to inherit medical.appointment. Since I'm a newbie in odoo development too.

**Comment by lasley** [comment]
@askz  Bit of history - this library has a lot of modules that were combined into super-modules of sorts. We've been working at breaking them up into their independent features, as well as isolate to what the modules are actually working with. You can really see this history by looking at the ones we haven't touched yet, like medical_invoice.

Is your existing code in your repo somewhere so I can take a look? I'd be happy to offer a bit of assistance.

**Comment by askz** [comment]
@lasley okay great. Thanks for this rewind.
I'll do a public release in the end of the week, I have to work on this a little. 
Thanks for your assistance guys!
